pkg/ingester/shipper.go

```diff
@@ -54,6 +55,11 @@ func newMetrics(reg prometheus.Registerer) *metrics {
 		Name: "thanos_shipper_upload_failures_total",
 		Help: "Total number of block upload failures",
 	})
+	m.lastSuccessfulUploadTime = promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+		Name: "mimir_ingester_shipper_bucket_last_successful_upload_time",
+		Help: "Second timestamp of the last successful TSDB block uploaded to the bucket.",
+	}, []string{"bucket"})
```

Comment:
No need to make it per bucket.

```suggestion
	})
```